### PR TITLE
chore(deps): upgrade axios to 1.6.0

### PR DESCRIPTION
**Pull request by elorzafe**
#### Description of changes

- upgrade axios version on `api-rest`
- upgrade typescript version to `5.0.2` and build scripts on `api-rest`, `api-graphql` and `api` packages similar to `core`

#### Issue #, if available

Axios version needs to be upgrade

#### Description of how you validated changes
- unit tests
- smoke test

#### Checklist

- [x] PR description included
- [x] `yarn test` passes
- [x] Tests are [changed or added](https://github.com/aws-amplify/amplify-js/blob/main/CONTRIBUTING.md#steps-towards-contributions)
- [ ] Relevant documentation is changed or added (and PR referenced)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by david-mcafee — DISMISSED** [review]
LGTM!

**Review by AllanZhengYP — APPROVED** [review]
The change looks good to me

File: packages/api/package.json [hunk 1]
```diff
@@ -48,7 +48,8 @@
 	},
 	"homepage": "https://aws-amplify.github.io/",
 	"devDependencies": {
-		"@types/zen-observable": "^0.8.0"
+		"@types/zen-observable": "^0.8.0",
+		"typescript": "5.0.2"
```
Comment (AllanZhengYP) [inline packages/api/package.json:52]: Can you do a quick check with the v6 [tsc-compatibility-test](https://github.com/aws-amplify/amplify-js/tree/main/scripts/tsc-compliance-test) to make sure we don't break interface consumption in older TSC?

**Review by AllanZhengYP — APPROVED** [review]
Looks good to me!

File: .github/workflows/push-preid-release.yml [hunk 2]
```diff
@@ -9,7 +9,7 @@ on:
   push:
     branches:
       # Change this to your branch name where "example-preid" corresponds to the preid you want your changes released on
-      - feat/example-preid-branch/main
+      - fix/v5-upgrade-axios
```
Comment (david-mcafee) [inline .github/workflows/push-preid-release.yml:12]: Remove?

**Review by david-mcafee — CHANGES_REQUESTED** [review]

File: packages/api-graphql/package.json [hunk 3]
```diff
@@ -19,10 +19,10 @@
...
     "test:size": "size-limit",
     "build-with-test": "npm test && npm run build",
-    "build:cjs": "node ./build es5 && webpack && webpack --config ./webpack.config.dev.js",
-    "build:esm": "node ./build es6",
-    "build:cjs:watch": "node ./build es5 --watch",
-    "build:esm:watch": "node ./build es6 --watch",
+    "build:cjs": "rimraf lib && tsc -m commonjs --outDir lib && webpack && webpack --config ./webpack.config.dev.js",
+		"build:esm": "rimraf lib-esm && tsc -m esnext --outDir lib-esm",
+		"build:cjs:watch": "rimraf lib && tsc -m commonjs --outDir lib --watch",
+		"build:esm:watch": "rimraf lib-esm && tsc -m esnext --outDir lib-esm --watch",
```
Comment (jimblanc) [inline packages/api-graphql/package.json:25]: Nit but tabbing error

File: packages/api-graphql/tsconfig.json [hunk 4]
```diff
@@ -0,0 +1,11 @@
+{
+	"extends": "../tsconfig.base.json",
+	"compilerOptions": {
+		"importHelpers": false,
```
Comment (jimblanc) [inline packages/api-graphql/tsconfig.json:4]: Should this be true to reduce bundle size?

File: packages/api-rest/package.json [hunk 5]
```diff
@@ -47,60 +47,60 @@
   ],
   "dependencies": {
     "@aws-amplify/core": "5.8.5",
-    "axios": "0.26.0",
+    "axios": "1.6.0",
```
Comment (jimblanc) [inline packages/api-rest/package.json:50]: I wonder if we want to use `^1.6.0` to improve flattening with customer apps when new versions are released 🤔

File: packages/api-rest/package.json [hunk 6]
```diff
@@ -19,10 +19,10 @@
...
     "test:size": "size-limit",
     "build-with-test": "npm test && npm run build",
-    "build:cjs": "node ./build es5 && webpack && webpack --config ./webpack.config.dev.js",
-    "build:esm": "node ./build es6",
-    "build:cjs:watch": "node ./build es5 --watch",
-    "build:esm:watch": "node ./build es6 --watch",
+    "build:cjs": "rimraf lib && tsc -m commonjs --outDir lib && webpack && webpack --config ./webpack.config.dev.js",
+		"build:esm": "rimraf lib-esm && tsc -m esnext --outDir lib-esm",
+		"build:cjs:watch": "rimraf lib && tsc -m commonjs --outDir lib --watch",
+		"build:esm:watch": "rimraf lib-esm && tsc -m esnext --outDir lib-esm --watch",
```
Comment (jimblanc) [inline packages/api-rest/package.json:25]: Nit: Tabbing error

File: packages/api-rest/tsconfig.json [hunk 7]
```diff
@@ -0,0 +1,10 @@
+{
+	"extends": "../tsconfig.base.json",
+	"compilerOptions": {
+		"importHelpers": false
```
Comment (jimblanc) [inline packages/api-rest/tsconfig.json:4]: Ditto

File: packages/api/tsconfig.json [hunk 8]
```diff
@@ -0,0 +1,11 @@
+{
+	"extends": "../tsconfig.base.json",
+	"compilerOptions": {
+		"importHelpers": false,
```
Comment (jimblanc) [inline packages/api/tsconfig.json:4]: Ditto

**Review by jimblanc — COMMENTED** [review]
Looks good, some nits & questions

File: packages/api-graphql/tsconfig.json (hunk 4 above)
Comment (elorzafe) [reply packages/api-graphql/tsconfig.json:4]: good call, let me update that

File: packages/api-rest/package.json (hunk 5 above)
Comment (elorzafe) [reply packages/api-rest/package.json:50]: The only concern with that is that could break something, specially since we are using a deprecated api

**Review by AllanZhengYP — APPROVED** [review]
Approving again. The bundle size increase reflects the Axios' own size increase from 0.26 to 1.6

**Review by stocaaro — DISMISSED** [review]

**Review by AllanZhengYP — APPROVED** [review]

File: .github/workflows/push-preid-release.yml (hunk 2 above)
Comment (elorzafe) [inline .github/workflows/push-preid-release.yml:12]:
```suggestion
      - feat/example-preid-branch/main
```

**Review by jimblanc — APPROVED** [review]

**Review by AllanZhengYP — APPROVED** [review]

**Pull request merged**